relay-general/derive/src/lib.rs

```diff
@@ -1033,3 +1041,54 @@ fn parse_variant_attributes(attrs: &[syn::Attribute]) -> VariantAttrs {
 fn is_newtype(variant: &synstructure::VariantInfo) -> bool {
     variant.bindings().len() == 1 && variant.bindings()[0].ast().ident.is_none()
 }
+
+fn parse_character_set(ident: &Ident, value: &str) -> TokenStream {
```

Comment:
This is the kind of function that is fun to write (always feeling smart when writing trivial parsers), but not great to look back at.